lxd/db/storage_volumes.go

```diff
@@ -1056,3 +1058,46 @@ WHERE storage_volumes.type = ? AND projects.name = ?
 
 	return volumes, nil
 }
+
+// GetStorageVolumeURIs returns the URIs of the storage volumes, specifying
+// target node if applicable
+func (c *ClusterTx) GetStorageVolumeURIs(project string) ([]string, error) {
+	code := cluster.EntityTypes["storage volume"]
+	volInfo, err := c.GetCustomVolumesInProject(project)
+	if err != nil {
+		return nil, err
+	}
+
+	uris := []string{}
+	for _, info := range volInfo {
+		uri := fmt.Sprintf(cluster.EntityURIs[code], info.PoolName, "custom", info.Name, project)
+		formChar := "&"
+		if project == "default" {
+			uri = strings.Split(uri, fmt.Sprintf("?project=%s", project))[0]
```

Comment:
@masnax please can you use the api.URL builder for this part

https://github.com/lxc/lxd/blob/master/shared/api/url.go

e.g. 

https://github.com/lxc/lxd/blob/6e377a8121ebe41f668183191c3bc9787bd35df1/lxd/network/driver_common.go#L1046

Reply:
So I have this branch that fixes up all the URI handling in the generator: https://github.com/masnax/lxd/tree/generator/uris

I haven't put it up as PR yet so as to not pollute the list of PRs, but I can do so if that would be better.

Reply:
Ah, okay, considering that as response for an earlier question too.